data/mesongenbuildreq.py

```diff
@@ -1,8 +1,8 @@
 import subprocess
 import json
 import sys
-deps_json = json.loads(subprocess.run([sys.argv[0], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout)
-deps = dict(zip([x['name'] for x in deps_json],[x['version'] for x in deps_json]))
+deps_json = json.loads(subprocess.run([sys.argv[1], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout)
+deps = dict(sorted(zip([x['name'] for x in deps_json],[x['version'] for x in deps_json])))
```

Comment:
python dicts have a randomized hash seed, so this might still produce non-deterministic output.